File: backend/src/main/java/de/neuefische/capstone/backend/model/EntryWithNoId.java
package de.neuefische.capstone.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@AllArgsConstructor
public class EntryWithNoId {

    private final String title;
    private final String description;
    private final LocalDate date;
    private final BigDecimal amount;
    private final Category category;
    private final Interval interval;
}

File: frontend/src/model/EntryWithNoId.ts
import {Interval} from "./Interval.ts";
import {Category} from "./Category.ts";
import {CostType} from "./CostType.ts";

export type EntryWithNoId = {
    title: string;
    description: string;
    date: string;
    amount: string;
    interval: Interval;
    category: Category;
    costType: CostType;
}